modules/oe_translation_poetry/tests/Functional/PoetryTranslationRequestTest.php

```diff
@@ -275,6 +287,302 @@ public function testUnprocessedJobHandling(): void {
+      if ($langcode === 'it') {
+        // Italian was refused.
+        $this->assertTrue($job->get('poetry_state')->isEmpty());
+        $this->assertTrue($job->get('poetry_request_date_updated')->isEmpty());
+        $this->assertEqual($job->getState(), Job::STATE_REJECTED);
+        $this->assertCount(1, $job->getMessages());
+        continue;
+      }
+
+      // The others were accepted.
+      $this->assertEqual($job->get('poetry_state')->value, PoetryTranslator::POETRY_STATUS_ONGOING);
+      $date = $job->get('poetry_request_date_updated')->date;
+      $this->assertEqual($date->format('d/m/Y'), '30/09/2019');
+      $this->assertEqual($job->getState(), Job::STATE_ACTIVE);
+      // Two messages are set on the job: date update and marking it as ongoing.
+      $this->assertCount(2, $job->getMessages());
+    }
+
+    // Send a notification cancelling one of the languages that were accepted.
+    $status_notification = $this->fixtureGenerator->statusNotification($identifier_info, 'ONG', [], [], [
+      [
+        'code' => 'DE',
+      ],
+    ]);
+
+    $this->performNotification($status_notification);
+
+    $this->jobStorage->resetCache();
+    /** @var \Drupal\tmgmt\JobInterface[] $jobs */
+    $jobs = $this->groupJobsByLanguage($this->jobStorage->loadMultiple());
+    foreach ($jobs as $langcode => $job) {
+      if ($langcode === 'fr') {
+        // French remained accepted.
+        $this->assertEqual($job->get('poetry_state')->value, PoetryTranslator::POETRY_STATUS_ONGOING);
+        $this->assertEqual($job->getState(), Job::STATE_ACTIVE);
+        continue;
+      }
+
+      // The others were cancelled.
+      $this->assertTrue($job->get('poetry_state')->isEmpty());
+      $this->assertTrue($job->get('poetry_request_date_updated')->isEmpty());
+      $this->assertEqual($job->getState(), Job::STATE_REJECTED);
+    }
+  }
+
+  /**
+   * Tests the handling of Poetry translation notifications.
+   */
+  public function testTranslationNotifications(): void {
+    $this->prepareRequestedJobs([
+      'title' => 'My node title',
+      'field_oe_demo_translatable_body' => 'My node body',
+    ], ['fr']);
+
+    $identifier_info = [
+      'code' => 'WEB',
+      'part' => '0',
+      'version' => '0',
+      'product' => 'TRA',
+      'number' => 3234,
+      'year' => 2010,
+    ];
+    $identifier = new Identifier();
+    foreach ($identifier_info as $name => $value) {
+      $identifier->offsetSet($name, $value);
+    }
+
+    // Accept the translations.
+    $status_notification = $this->fixtureGenerator->statusNotification($identifier_info, 'ONG',
+      [
+        [
+          'code' => 'FR',
+          'date' => '30/08/2019 23:59',
+          'accepted_date' => '30/09/2019 23:59',
+        ],
+      ]);
+
+    $this->performNotification($status_notification);
+
+    // Send the translations for each job.
+    $jobs = $this->groupJobsByLanguage($this->jobStorage->loadMultiple());
+
+    foreach ($jobs as $job) {
```

Comment:
We are only testing a single translation so is there need to do this foreach loops? The whole loop is then done again on #434 so removing it would simplify the whole test method a lot.

Reply:
Will do.